plugins/modules/vmware_host_facts.py

```diff
@@ -296,9 +302,14 @@ def get_cpu_facts(self):
         }
 
     def get_memory_facts(self):
+        memory_size = self.host.hardware.memorySize
+        overall_memory = 0
+        if self.host.summary.quickStats.overallMemoryUsage:
```

Comment:
Just a quick thought. If the host is down, `ansible_memfree_mb` is meaningless. I feel like we should return `None` instead.